src/Illuminate/Auth/Middleware/EnsureEmailIsVerified.php

```diff
@@ -13,14 +13,18 @@ class EnsureEmailIsVerified
      *
      * @param  \Illuminate\Http\Request  $request
      * @param  \Closure  $next
-     * @return \Illuminate\Http\Response
+     * @return \Illuminate\Http\Response|\Illuminate\Http\RedirectResponse
      */
     public function handle($request, Closure $next)
     {
         if (! $request->user() ||
             ($request->user() instanceof MustVerifyEmail &&
             ! $request->user()->hasVerifiedEmail())) {
-            return Redirect::route('verification.notice');
+            if ($request->expectsJson()) {
+                abort(403, 'Your email address is not verified.');
+            } else {
+                return Redirect::route('verification.notice');
```

Comment:
You dont need the "else" here. Just return since the `if` will catch the other condition.

Reply:
You are, of course, correct! I changed it.